### PR TITLE
#123/Create controllers to fetch the user avatar

### PR DESCRIPTION
**Pull request by gbudau**
GET [/api/v1/users/avatar](http://localhost:3000/api#/users/UserController_getCurrentUserAvatar) -> get the avatar of the currently authenticated user
GET [/api/v1/users/{uuid}/avatar](http://localhost:3000/api#/users/UserController_getAvatar) -> get the avatar of the user with uuid

image

Subtask of #123

### CODE REVIEW

**Comment by Damrod** [comment]
Do we need, again to remove the docker volumes?
```
ft_transcendence-transcendence-app-1  | src/shared/local-file/local-file.interceptor.ts:5:29 - error TS7016: Could not find a declaration file for module 'multer'. '/transcendence-app/node_modules/multer/index.js' implicitly has an 'any' type.
ft_transcendence-transcendence-app-1  |   Try `npm i --save-dev @types/multer` if it exists or add a new declaration (.d.ts) file containing `declare module 'multer';`
ft_transcendence-transcendence-app-1  | 
ft_transcendence-transcendence-app-1  | 5 import { diskStorage } from 'multer';
ft_transcendence-transcendence-app-1  |                               ~~~~~~~~
ft_transcendence-transcendence-app-1  | 
ft_transcendence-transcendence-app-1  | [12:51:15 PM] Found 1 error. Watching for file changes.
ft_transcendence-transcendence-app-1  | 
```

**Comment by gbudau** [comment]
I think only the anonymous volumes, the `-V` flag

```
docker compose up --build -d -V
```

**Comment by Damrod** [comment]
Now it finally works, but I had to manually remove all volumes, all containers, and rerun them all.

File: transcendence-app/src/user/user.controller.ts [hunk 1]
```diff
@@ -22,6 +23,7 @@ import {
   ApiForbiddenResponse,
   ApiNotFoundResponse,
   ApiOkResponse,
+  ApiProduces,
```
Comment (Damrod) [inline transcendence-app/src/user/user.controller.ts:26]: It's a bit outside of the scope of this PR, but I have just noticed we don't use the `AvatarFileInterceptor` . Are we planning to use it in the future?

Comment (Damrod) [reply transcendence-app/src/user/user.controller.ts:26]: In fact we are not using the local File interceptor (which seems strange to me...)
What am I missing?

Comment (gbudau) [reply transcendence-app/src/user/user.controller.ts:26]: You are right. We don't use file interceptors in this PR. We use the `AvatarFileInterceptor` in #126 to save the avatar on disk and check mime type and size.

Comment (Damrod) [reply transcendence-app/src/user/user.controller.ts:26]: It's PR !126 for ticket #123. That's confusing.

**Comment by gbudau** [comment]
Thank you!

**Review by fizcamposancos — COMMENTED** [review]
LGTM!

**Pull request merged**